// @ziqni-tech/gamification-api-client/dist/model/ContestFilter.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ContestStatus = _interopRequireDefault(require("./ContestStatus"));

var _QuerySortBy = _interopRequireDefault(require("./QuerySortBy"));

var _RangeQuery = _interopRequireDefault(require("./RangeQuery"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The ContestFilter model module.
 * @module model/ContestFilter
 * @version 0.0.1
 */
var ContestFilter = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ContestFilter</code>.
   * Filter to trim results from list of Contets
   * @alias module:model/ContestFilter
   */
  function ContestFilter() {
    _classCallCheck(this, ContestFilter);

    ContestFilter.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ContestFilter, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ContestFilter</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ContestFilter} obj Optional instance to populate.
     * @return {module:model/ContestFilter} The populated <code>ContestFilter</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ContestFilter();

        if (data.hasOwnProperty('productIds')) {
          obj['productIds'] = _ApiClient["default"].convertToType(data['productIds'], ['String']);
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('startDate')) {
          obj['startDate'] = _RangeQuery["default"].constructFromObject(data['startDate']);
        }

        if (data.hasOwnProperty('endDate')) {
          obj['endDate'] = _RangeQuery["default"].constructFromObject(data['endDate']);
        }

        if (data.hasOwnProperty('sortBy')) {
          obj['sortBy'] = _QuerySortBy["default"].constructFromObject(data['sortBy']);
        }

        if (data.hasOwnProperty('competitionIds')) {
          obj['competitionIds'] = _ApiClient["default"].convertToType(data['competitionIds'], ['String']);
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _ApiClient["default"].convertToType(data['status'], [_ContestStatus["default"]]);
        }

        if (data.hasOwnProperty('skip')) {
          obj['skip'] = _ApiClient["default"].convertToType(data['skip'], 'Number');
        }

        if (data.hasOwnProperty('limit')) {
          obj['limit'] = _ApiClient["default"].convertToType(data['limit'], 'Number');
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return ContestFilter;
}();
/**
 * @member {Array.<String>} productIds
 */


ContestFilter.prototype['productIds'] = undefined;
/**
 * @member {Array.<String>} tags
 */

ContestFilter.prototype['tags'] = undefined;
/**
 * @member {module:model/RangeQuery} startDate
 */

ContestFilter.prototype['startDate'] = undefined;
/**
 * @member {module:model/RangeQuery} endDate
 */

ContestFilter.prototype['endDate'] = undefined;
/**
 * @member {module:model/QuerySortBy} sortBy
 */

ContestFilter.prototype['sortBy'] = undefined;
/**
 * @member {Array.<String>} competitionIds
 */

ContestFilter.prototype['competitionIds'] = undefined;
/**
 * @member {Array.<module:model/ContestStatus>} status
 */

ContestFilter.prototype['status'] = undefined;
/**
 * @member {Number} skip
 */

ContestFilter.prototype['skip'] = undefined;
/**
 * @member {Number} limit
 */

ContestFilter.prototype['limit'] = undefined;
/**
 * @member {Array.<String>} constraints
 */

ContestFilter.prototype['constraints'] = undefined;
var _default = ContestFilter;
exports["default"] = _default;
