// @ziqni-tech/gamification-api-client/dist/model/MessageAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Dependancy = _interopRequireDefault(require("./Dependancy"));

var _Scheduling = _interopRequireDefault(require("./Scheduling"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The MessageAllOf model module.
 * @module model/MessageAllOf
 * @version 0.0.1
 */
var MessageAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MessageAllOf</code>.
   * @alias module:model/MessageAllOf
   */
  function MessageAllOf() {
    _classCallCheck(this, MessageAllOf);

    MessageAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(MessageAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>MessageAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MessageAllOf} obj Optional instance to populate.
     * @return {module:model/MessageAllOf} The populated <code>MessageAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MessageAllOf();

        if (data.hasOwnProperty('icon')) {
          obj['icon'] = _ApiClient["default"].convertToType(data['icon'], 'String');
        }

        if (data.hasOwnProperty('jsonClass')) {
          obj['jsonClass'] = _ApiClient["default"].convertToType(data['jsonClass'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('termsAndConditions')) {
          obj['termsAndConditions'] = _ApiClient["default"].convertToType(data['termsAndConditions'], 'String');
        }

        if (data.hasOwnProperty('constraints')) {
          obj['constraints'] = _ApiClient["default"].convertToType(data['constraints'], ['String']);
        }

        if (data.hasOwnProperty('maxNumberOfIssues')) {
          obj['maxNumberOfIssues'] = _ApiClient["default"].convertToType(data['maxNumberOfIssues'], 'Number');
        }

        if (data.hasOwnProperty('scheduling')) {
          obj['scheduling'] = _Scheduling["default"].constructFromObject(data['scheduling']);
        }

        if (data.hasOwnProperty('dependantOn')) {
          obj['dependantOn'] = _Dependancy["default"].constructFromObject(data['dependantOn']);
        }
      }

      return obj;
    }
  }]);

  return MessageAllOf;
}();
/**
 * An Icon id that has been pre uploaded to the system to display for Achievement
 * @member {String} icon
 */


MessageAllOf.prototype['icon'] = undefined;
/**
 * @member {String} jsonClass
 */

MessageAllOf.prototype['jsonClass'] = undefined;
/**
 * @member {String} name
 */

MessageAllOf.prototype['name'] = undefined;
/**
 * @member {String} description
 */

MessageAllOf.prototype['description'] = undefined;
/**
 * @member {String} termsAndConditions
 */

MessageAllOf.prototype['termsAndConditions'] = undefined;
/**
 * Additional constraints, if set means true
 * @member {Array.<String>} constraints
 */

MessageAllOf.prototype['constraints'] = undefined;
/**
 * Maximum number of issued achievements
 * @member {Number} maxNumberOfIssues
 */

MessageAllOf.prototype['maxNumberOfIssues'] = undefined;
/**
 * @member {module:model/Scheduling} scheduling
 */

MessageAllOf.prototype['scheduling'] = undefined;
/**
 * @member {module:model/Dependancy} dependantOn
 */

MessageAllOf.prototype['dependantOn'] = undefined;
var _default = MessageAllOf;
exports["default"] = _default;
