// @ziqni-tech/gamification-api-client/dist/model/ProductReduced.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The ProductReduced model module.
 * @module model/ProductReduced
 * @version 0.0.1
 */
var ProductReduced = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ProductReduced</code>.
   * @alias module:model/ProductReduced
   */
  function ProductReduced() {
    _classCallCheck(this, ProductReduced);

    ProductReduced.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ProductReduced, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ProductReduced</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ProductReduced} obj Optional instance to populate.
     * @return {module:model/ProductReduced} The populated <code>ProductReduced</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ProductReduced();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('productRefId')) {
          obj['productRefId'] = _ApiClient["default"].convertToType(data['productRefId'], 'String');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return ProductReduced;
}();
/**
 * The name of the product
 * @member {String} name
 */


ProductReduced.prototype['name'] = undefined;
/**
 * The reference to this product in your system. The reference identifier can not be changed after the product has been created
 * @member {String} productRefId
 */

ProductReduced.prototype['productRefId'] = undefined;
/**
 * A list of Strings used to tag products with taxonomy terms
 * @member {Array.<String>} tags
 */

ProductReduced.prototype['tags'] = undefined;
var _default = ProductReduced;
exports["default"] = _default;
