// @ziqni-tech/gamification-api-client/dist/model/ProxyResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Error = _interopRequireDefault(require("./Error"));

var _Proxy = _interopRequireDefault(require("./Proxy"));

var _ResponseMeta = _interopRequireDefault(require("./ResponseMeta"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The ProxyResponse model module.
 * @module model/ProxyResponse
 * @version 0.0.1
 */
var ProxyResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ProxyResponse</code>.
   * @alias module:model/ProxyResponse
   */
  function ProxyResponse() {
    _classCallCheck(this, ProxyResponse);

    ProxyResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ProxyResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ProxyResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ProxyResponse} obj Optional instance to populate.
     * @return {module:model/ProxyResponse} The populated <code>ProxyResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ProxyResponse();

        if (data.hasOwnProperty('meta')) {
          obj['meta'] = _ResponseMeta["default"].constructFromObject(data['meta']);
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_Proxy["default"]]);
        }

        if (data.hasOwnProperty('errors')) {
          obj['errors'] = _ApiClient["default"].convertToType(data['errors'], [_Error["default"]]);
        }
      }

      return obj;
    }
  }]);

  return ProxyResponse;
}();
/**
 * @member {module:model/ResponseMeta} meta
 */


ProxyResponse.prototype['meta'] = undefined;
/**
 * @member {Array.<module:model/Proxy>} data
 */

ProxyResponse.prototype['data'] = undefined;
/**
 * @member {Array.<module:model/Error>} errors
 */

ProxyResponse.prototype['errors'] = undefined;
var _default = ProxyResponse;
exports["default"] = _default;
