// @ziqni-tech/gamification-api-client/dist/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Achievement", {
  enumerable: true,
  get: function get() {
    return _Achievement["default"];
  }
});
Object.defineProperty(exports, "AchievementAllOf", {
  enumerable: true,
  get: function get() {
    return _AchievementAllOf["default"];
  }
});
Object.defineProperty(exports, "AchievementFilter", {
  enumerable: true,
  get: function get() {
    return _AchievementFilter["default"];
  }
});
Object.defineProperty(exports, "AchievementRequest", {
  enumerable: true,
  get: function get() {
    return _AchievementRequest["default"];
  }
});
Object.defineProperty(exports, "AchievementResponse", {
  enumerable: true,
  get: function get() {
    return _AchievementResponse["default"];
  }
});
Object.defineProperty(exports, "AchievementsApi", {
  enumerable: true,
  get: function get() {
    return _AchievementsApi["default"];
  }
});
Object.defineProperty(exports, "ApiClient", {
  enumerable: true,
  get: function get() {
    return _ApiClient["default"];
  }
});
Object.defineProperty(exports, "Award", {
  enumerable: true,
  get: function get() {
    return _Award["default"];
  }
});
Object.defineProperty(exports, "AwardFilter", {
  enumerable: true,
  get: function get() {
    return _AwardFilter["default"];
  }
});
Object.defineProperty(exports, "AwardRequest", {
  enumerable: true,
  get: function get() {
    return _AwardRequest["default"];
  }
});
Object.defineProperty(exports, "AwardResponse", {
  enumerable: true,
  get: function get() {
    return _AwardResponse["default"];
  }
});
Object.defineProperty(exports, "AwardsApi", {
  enumerable: true,
  get: function get() {
    return _AwardsApi["default"];
  }
});
Object.defineProperty(exports, "Competition", {
  enumerable: true,
  get: function get() {
    return _Competition["default"];
  }
});
Object.defineProperty(exports, "CompetitionAllOf", {
  enumerable: true,
  get: function get() {
    return _CompetitionAllOf["default"];
  }
});
Object.defineProperty(exports, "CompetitionFilter", {
  enumerable: true,
  get: function get() {
    return _CompetitionFilter["default"];
  }
});
Object.defineProperty(exports, "CompetitionRequest", {
  enumerable: true,
  get: function get() {
    return _CompetitionRequest["default"];
  }
});
Object.defineProperty(exports, "CompetitionResponse", {
  enumerable: true,
  get: function get() {
    return _CompetitionResponse["default"];
  }
});
Object.defineProperty(exports, "CompetitionStatus", {
  enumerable: true,
  get: function get() {
    return _CompetitionStatus["default"];
  }
});
Object.defineProperty(exports, "CompetitionType", {
  enumerable: true,
  get: function get() {
    return _CompetitionType["default"];
  }
});
Object.defineProperty(exports, "CompetitionsApi", {
  enumerable: true,
  get: function get() {
    return _CompetitionsApi["default"];
  }
});
Object.defineProperty(exports, "Contest", {
  enumerable: true,
  get: function get() {
    return _Contest["default"];
  }
});
Object.defineProperty(exports, "ContestAllOf", {
  enumerable: true,
  get: function get() {
    return _ContestAllOf["default"];
  }
});
Object.defineProperty(exports, "ContestFilter", {
  enumerable: true,
  get: function get() {
    return _ContestFilter["default"];
  }
});
Object.defineProperty(exports, "ContestReduced", {
  enumerable: true,
  get: function get() {
    return _ContestReduced["default"];
  }
});
Object.defineProperty(exports, "ContestRequest", {
  enumerable: true,
  get: function get() {
    return _ContestRequest["default"];
  }
});
Object.defineProperty(exports, "ContestResponse", {
  enumerable: true,
  get: function get() {
    return _ContestResponse["default"];
  }
});
Object.defineProperty(exports, "ContestStatus", {
  enumerable: true,
  get: function get() {
    return _ContestStatus["default"];
  }
});
Object.defineProperty(exports, "ContestsApi", {
  enumerable: true,
  get: function get() {
    return _ContestsApi["default"];
  }
});
Object.defineProperty(exports, "Dependancy", {
  enumerable: true,
  get: function get() {
    return _Dependancy["default"];
  }
});
Object.defineProperty(exports, "DependantOn", {
  enumerable: true,
  get: function get() {
    return _DependantOn["default"];
  }
});
Object.defineProperty(exports, "EntityType", {
  enumerable: true,
  get: function get() {
    return _EntityType["default"];
  }
});
Object.defineProperty(exports, "Error", {
  enumerable: true,
  get: function get() {
    return _Error["default"];
  }
});
Object.defineProperty(exports, "EventRefType", {
  enumerable: true,
  get: function get() {
    return _EventRefType["default"];
  }
});
Object.defineProperty(exports, "FieldType", {
  enumerable: true,
  get: function get() {
    return _FieldType["default"];
  }
});
Object.defineProperty(exports, "File", {
  enumerable: true,
  get: function get() {
    return _File["default"];
  }
});
Object.defineProperty(exports, "FileRequest", {
  enumerable: true,
  get: function get() {
    return _FileRequest["default"];
  }
});
Object.defineProperty(exports, "FileResponse", {
  enumerable: true,
  get: function get() {
    return _FileResponse["default"];
  }
});
Object.defineProperty(exports, "FilesApi", {
  enumerable: true,
  get: function get() {
    return _FilesApi["default"];
  }
});
Object.defineProperty(exports, "Member", {
  enumerable: true,
  get: function get() {
    return _Member["default"];
  }
});
Object.defineProperty(exports, "MemberAllOf", {
  enumerable: true,
  get: function get() {
    return _MemberAllOf["default"];
  }
});
Object.defineProperty(exports, "MemberOptinRequest", {
  enumerable: true,
  get: function get() {
    return _MemberOptinRequest["default"];
  }
});
Object.defineProperty(exports, "MemberRequest", {
  enumerable: true,
  get: function get() {
    return _MemberRequest["default"];
  }
});
Object.defineProperty(exports, "MemberResponse", {
  enumerable: true,
  get: function get() {
    return _MemberResponse["default"];
  }
});
Object.defineProperty(exports, "MemberSessionRequest", {
  enumerable: true,
  get: function get() {
    return _MemberSessionRequest["default"];
  }
});
Object.defineProperty(exports, "MemberSessionResponse", {
  enumerable: true,
  get: function get() {
    return _MemberSessionResponse["default"];
  }
});
Object.defineProperty(exports, "MemberType", {
  enumerable: true,
  get: function get() {
    return _MemberType["default"];
  }
});
Object.defineProperty(exports, "MembersApi", {
  enumerable: true,
  get: function get() {
    return _MembersApi["default"];
  }
});
Object.defineProperty(exports, "Message", {
  enumerable: true,
  get: function get() {
    return _Message["default"];
  }
});
Object.defineProperty(exports, "MessageRequest", {
  enumerable: true,
  get: function get() {
    return _MessageRequest["default"];
  }
});
Object.defineProperty(exports, "MessageResponse", {
  enumerable: true,
  get: function get() {
    return _MessageResponse["default"];
  }
});
Object.defineProperty(exports, "MessageStatus", {
  enumerable: true,
  get: function get() {
    return _MessageStatus["default"];
  }
});
Object.defineProperty(exports, "MessageType", {
  enumerable: true,
  get: function get() {
    return _MessageType["default"];
  }
});
Object.defineProperty(exports, "MessagesApi", {
  enumerable: true,
  get: function get() {
    return _MessagesApi["default"];
  }
});
Object.defineProperty(exports, "ModelDefault", {
  enumerable: true,
  get: function get() {
    return _ModelDefault["default"];
  }
});
Object.defineProperty(exports, "OptinAction", {
  enumerable: true,
  get: function get() {
    return _OptinAction["default"];
  }
});
Object.defineProperty(exports, "OptionalModelFields", {
  enumerable: true,
  get: function get() {
    return _OptionalModelFields["default"];
  }
});
Object.defineProperty(exports, "ProductReduced", {
  enumerable: true,
  get: function get() {
    return _ProductReduced["default"];
  }
});
Object.defineProperty(exports, "Proxy", {
  enumerable: true,
  get: function get() {
    return _Proxy["default"];
  }
});
Object.defineProperty(exports, "ProxyApi", {
  enumerable: true,
  get: function get() {
    return _ProxyApi["default"];
  }
});
Object.defineProperty(exports, "ProxyRequest", {
  enumerable: true,
  get: function get() {
    return _ProxyRequest["default"];
  }
});
Object.defineProperty(exports, "ProxyResponse", {
  enumerable: true,
  get: function get() {
    return _ProxyResponse["default"];
  }
});
Object.defineProperty(exports, "QueryMultipleFields", {
  enumerable: true,
  get: function get() {
    return _QueryMultipleFields["default"];
  }
});
Object.defineProperty(exports, "QuerySortBy", {
  enumerable: true,
  get: function get() {
    return _QuerySortBy["default"];
  }
});
Object.defineProperty(exports, "RangeQuery", {
  enumerable: true,
  get: function get() {
    return _RangeQuery["default"];
  }
});
Object.defineProperty(exports, "RankingStrategy", {
  enumerable: true,
  get: function get() {
    return _RankingStrategy["default"];
  }
});
Object.defineProperty(exports, "ResponseMeta", {
  enumerable: true,
  get: function get() {
    return _ResponseMeta["default"];
  }
});
Object.defineProperty(exports, "Reward", {
  enumerable: true,
  get: function get() {
    return _Reward["default"];
  }
});
Object.defineProperty(exports, "RewardReduced", {
  enumerable: true,
  get: function get() {
    return _RewardReduced["default"];
  }
});
Object.defineProperty(exports, "RewardRequest", {
  enumerable: true,
  get: function get() {
    return _RewardRequest["default"];
  }
});
Object.defineProperty(exports, "RewardResponse", {
  enumerable: true,
  get: function get() {
    return _RewardResponse["default"];
  }
});
Object.defineProperty(exports, "RewardTypeReduced", {
  enumerable: true,
  get: function get() {
    return _RewardTypeReduced["default"];
  }
});
Object.defineProperty(exports, "RewardsApi", {
  enumerable: true,
  get: function get() {
    return _RewardsApi["default"];
  }
});
Object.defineProperty(exports, "RoundType", {
  enumerable: true,
  get: function get() {
    return _RoundType["default"];
  }
});
Object.defineProperty(exports, "Rule", {
  enumerable: true,
  get: function get() {
    return _Rule["default"];
  }
});
Object.defineProperty(exports, "RuleArgument", {
  enumerable: true,
  get: function get() {
    return _RuleArgument["default"];
  }
});
Object.defineProperty(exports, "RuleExpression", {
  enumerable: true,
  get: function get() {
    return _RuleExpression["default"];
  }
});
Object.defineProperty(exports, "RuleRequest", {
  enumerable: true,
  get: function get() {
    return _RuleRequest["default"];
  }
});
Object.defineProperty(exports, "RuleResponse", {
  enumerable: true,
  get: function get() {
    return _RuleResponse["default"];
  }
});
Object.defineProperty(exports, "RuleRoutine", {
  enumerable: true,
  get: function get() {
    return _RuleRoutine["default"];
  }
});
Object.defineProperty(exports, "RulesApi", {
  enumerable: true,
  get: function get() {
    return _RulesApi["default"];
  }
});
Object.defineProperty(exports, "ScheduleType", {
  enumerable: true,
  get: function get() {
    return _ScheduleType["default"];
  }
});
Object.defineProperty(exports, "Scheduling", {
  enumerable: true,
  get: function get() {
    return _Scheduling["default"];
  }
});
Object.defineProperty(exports, "ScoringStrategy", {
  enumerable: true,
  get: function get() {
    return _ScoringStrategy["default"];
  }
});
Object.defineProperty(exports, "SortOrder", {
  enumerable: true,
  get: function get() {
    return _SortOrder["default"];
  }
});
Object.defineProperty(exports, "Strategy", {
  enumerable: true,
  get: function get() {
    return _Strategy["default"];
  }
});
Object.defineProperty(exports, "StrategyType", {
  enumerable: true,
  get: function get() {
    return _StrategyType["default"];
  }
});
Object.defineProperty(exports, "Translation", {
  enumerable: true,
  get: function get() {
    return _Translation["default"];
  }
});
Object.defineProperty(exports, "TranslationEntry", {
  enumerable: true,
  get: function get() {
    return _TranslationEntry["default"];
  }
});
Object.defineProperty(exports, "TranslationRequest", {
  enumerable: true,
  get: function get() {
    return _TranslationRequest["default"];
  }
});
Object.defineProperty(exports, "TranslationResponse", {
  enumerable: true,
  get: function get() {
    return _TranslationResponse["default"];
  }
});

var _ApiClient = _interopRequireDefault(require("./ApiClient"));

var _Achievement = _interopRequireDefault(require("./model/Achievement"));

var _AchievementAllOf = _interopRequireDefault(require("./model/AchievementAllOf"));

var _AchievementFilter = _interopRequireDefault(require("./model/AchievementFilter"));

var _AchievementRequest = _interopRequireDefault(require("./model/AchievementRequest"));

var _AchievementResponse = _interopRequireDefault(require("./model/AchievementResponse"));

var _Award = _interopRequireDefault(require("./model/Award"));

var _AwardFilter = _interopRequireDefault(require("./model/AwardFilter"));

var _AwardRequest = _interopRequireDefault(require("./model/AwardRequest"));

var _AwardResponse = _interopRequireDefault(require("./model/AwardResponse"));

var _Competition = _interopRequireDefault(require("./model/Competition"));

var _CompetitionAllOf = _interopRequireDefault(require("./model/CompetitionAllOf"));

var _CompetitionFilter = _interopRequireDefault(require("./model/CompetitionFilter"));

var _CompetitionRequest = _interopRequireDefault(require("./model/CompetitionRequest"));

var _CompetitionResponse = _interopRequireDefault(require("./model/CompetitionResponse"));

var _CompetitionStatus = _interopRequireDefault(require("./model/CompetitionStatus"));

var _CompetitionType = _interopRequireDefault(require("./model/CompetitionType"));

var _Contest = _interopRequireDefault(require("./model/Contest"));

var _ContestAllOf = _interopRequireDefault(require("./model/ContestAllOf"));

var _ContestFilter = _interopRequireDefault(require("./model/ContestFilter"));

var _ContestReduced = _interopRequireDefault(require("./model/ContestReduced"));

var _ContestRequest = _interopRequireDefault(require("./model/ContestRequest"));

var _ContestResponse = _interopRequireDefault(require("./model/ContestResponse"));

var _ContestStatus = _interopRequireDefault(require("./model/ContestStatus"));

var _Dependancy = _interopRequireDefault(require("./model/Dependancy"));

var _DependantOn = _interopRequireDefault(require("./model/DependantOn"));

var _EntityType = _interopRequireDefault(require("./model/EntityType"));

var _Error = _interopRequireDefault(require("./model/Error"));

var _EventRefType = _interopRequireDefault(require("./model/EventRefType"));

var _FieldType = _interopRequireDefault(require("./model/FieldType"));

var _File = _interopRequireDefault(require("./model/File"));

var _FileRequest = _interopRequireDefault(require("./model/FileRequest"));

var _FileResponse = _interopRequireDefault(require("./model/FileResponse"));

var _Member = _interopRequireDefault(require("./model/Member"));

var _MemberAllOf = _interopRequireDefault(require("./model/MemberAllOf"));

var _MemberOptinRequest = _interopRequireDefault(require("./model/MemberOptinRequest"));

var _MemberRequest = _interopRequireDefault(require("./model/MemberRequest"));

var _MemberResponse = _interopRequireDefault(require("./model/MemberResponse"));

var _MemberSessionRequest = _interopRequireDefault(require("./model/MemberSessionRequest"));

var _MemberSessionResponse = _interopRequireDefault(require("./model/MemberSessionResponse"));

var _MemberType = _interopRequireDefault(require("./model/MemberType"));

var _Message = _interopRequireDefault(require("./model/Message"));

var _MessageRequest = _interopRequireDefault(require("./model/MessageRequest"));

var _MessageResponse = _interopRequireDefault(require("./model/MessageResponse"));

var _MessageStatus = _interopRequireDefault(require("./model/MessageStatus"));

var _MessageType = _interopRequireDefault(require("./model/MessageType"));

var _ModelDefault = _interopRequireDefault(require("./model/ModelDefault"));

var _OptinAction = _interopRequireDefault(require("./model/OptinAction"));

var _OptionalModelFields = _interopRequireDefault(require("./model/OptionalModelFields"));

var _ProductReduced = _interopRequireDefault(require("./model/ProductReduced"));

var _Proxy = _interopRequireDefault(require("./model/Proxy"));

var _ProxyRequest = _interopRequireDefault(require("./model/ProxyRequest"));

var _ProxyResponse = _interopRequireDefault(require("./model/ProxyResponse"));

var _QueryMultipleFields = _interopRequireDefault(require("./model/QueryMultipleFields"));

var _QuerySortBy = _interopRequireDefault(require("./model/QuerySortBy"));

var _RangeQuery = _interopRequireDefault(require("./model/RangeQuery"));

var _RankingStrategy = _interopRequireDefault(require("./model/RankingStrategy"));

var _ResponseMeta = _interopRequireDefault(require("./model/ResponseMeta"));

var _Reward = _interopRequireDefault(require("./model/Reward"));

var _RewardReduced = _interopRequireDefault(require("./model/RewardReduced"));

var _RewardRequest = _interopRequireDefault(require("./model/RewardRequest"));

var _RewardResponse = _interopRequireDefault(require("./model/RewardResponse"));

var _RewardTypeReduced = _interopRequireDefault(require("./model/RewardTypeReduced"));

var _RoundType = _interopRequireDefault(require("./model/RoundType"));

var _Rule = _interopRequireDefault(require("./model/Rule"));

var _RuleArgument = _interopRequireDefault(require("./model/RuleArgument"));

var _RuleExpression = _interopRequireDefault(require("./model/RuleExpression"));

var _RuleRequest = _interopRequireDefault(require("./model/RuleRequest"));

var _RuleResponse = _interopRequireDefault(require("./model/RuleResponse"));

var _RuleRoutine = _interopRequireDefault(require("./model/RuleRoutine"));

var _ScheduleType = _interopRequireDefault(require("./model/ScheduleType"));

var _Scheduling = _interopRequireDefault(require("./model/Scheduling"));

var _ScoringStrategy = _interopRequireDefault(require("./model/ScoringStrategy"));

var _SortOrder = _interopRequireDefault(require("./model/SortOrder"));

var _Strategy = _interopRequireDefault(require("./model/Strategy"));

var _StrategyType = _interopRequireDefault(require("./model/StrategyType"));

var _Translation = _interopRequireDefault(require("./model/Translation"));

var _TranslationEntry = _interopRequireDefault(require("./model/TranslationEntry"));

var _TranslationRequest = _interopRequireDefault(require("./model/TranslationRequest"));

var _TranslationResponse = _interopRequireDefault(require("./model/TranslationResponse"));

var _AchievementsApi = _interopRequireDefault(require("./api/AchievementsApi"));

var _AwardsApi = _interopRequireDefault(require("./api/AwardsApi"));

var _CompetitionsApi = _interopRequireDefault(require("./api/CompetitionsApi"));

var _ContestsApi = _interopRequireDefault(require("./api/ContestsApi"));

var _FilesApi = _interopRequireDefault(require("./api/FilesApi"));

var _MembersApi = _interopRequireDefault(require("./api/MembersApi"));

var _MessagesApi = _interopRequireDefault(require("./api/MessagesApi"));

var _ProxyApi = _interopRequireDefault(require("./api/ProxyApi"));

var _RewardsApi = _interopRequireDefault(require("./api/RewardsApi"));

var _RulesApi = _interopRequireDefault(require("./api/RulesApi"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
