// @ziqni-tech/member-api-client/dist/model/CompetitionFilter.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CompetitionStatus = _interopRequireDefault(require("./CompetitionStatus"));

var _CompetitionType = _interopRequireDefault(require("./CompetitionType"));

var _QuerySortBy = _interopRequireDefault(require("./QuerySortBy"));

var _RangeQuery = _interopRequireDefault(require("./RangeQuery"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The CompetitionFilter model module.
 * @module model/CompetitionFilter
 * @version 0.0.1
 */
var CompetitionFilter = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CompetitionFilter</code>.
   * @alias module:model/CompetitionFilter
   */
  function CompetitionFilter() {
    _classCallCheck(this, CompetitionFilter);

    CompetitionFilter.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CompetitionFilter, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>CompetitionFilter</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CompetitionFilter} obj Optional instance to populate.
     * @return {module:model/CompetitionFilter} The populated <code>CompetitionFilter</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CompetitionFilter();

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('startDate')) {
          obj['startDate'] = _RangeQuery["default"].constructFromObject(data['startDate']);
        }

        if (data.hasOwnProperty('endDate')) {
          obj['endDate'] = _RangeQuery["default"].constructFromObject(data['endDate']);
        }

        if (data.hasOwnProperty('productIds')) {
          obj['productIds'] = _ApiClient["default"].convertToType(data['productIds'], ['String']);
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _ApiClient["default"].convertToType(data['status'], [_CompetitionStatus["default"]]);
        }

        if (data.hasOwnProperty('competitionType')) {
          obj['competitionType'] = _ApiClient["default"].convertToType(data['competitionType'], [_CompetitionType["default"]]);
        }

        if (data.hasOwnProperty('sortBy')) {
          obj['sortBy'] = _ApiClient["default"].convertToType(data['sortBy'], [_QuerySortBy["default"]]);
        }

        if (data.hasOwnProperty('ids')) {
          obj['ids'] = _ApiClient["default"].convertToType(data['ids'], ['String']);
        }

        if (data.hasOwnProperty('skip')) {
          obj['skip'] = _ApiClient["default"].convertToType(data['skip'], 'Number');
        }

        if (data.hasOwnProperty('limit')) {
          obj['limit'] = _ApiClient["default"].convertToType(data['limit'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return CompetitionFilter;
}();
/**
 * @member {Array.<String>} tags
 */


CompetitionFilter.prototype['tags'] = undefined;
/**
 * @member {module:model/RangeQuery} startDate
 */

CompetitionFilter.prototype['startDate'] = undefined;
/**
 * @member {module:model/RangeQuery} endDate
 */

CompetitionFilter.prototype['endDate'] = undefined;
/**
 * @member {Array.<String>} productIds
 */

CompetitionFilter.prototype['productIds'] = undefined;
/**
 * @member {Array.<module:model/CompetitionStatus>} status
 */

CompetitionFilter.prototype['status'] = undefined;
/**
 * @member {Array.<module:model/CompetitionType>} competitionType
 */

CompetitionFilter.prototype['competitionType'] = undefined;
/**
 * @member {Array.<module:model/QuerySortBy>} sortBy
 */

CompetitionFilter.prototype['sortBy'] = undefined;
/**
 * @member {Array.<String>} ids
 */

CompetitionFilter.prototype['ids'] = undefined;
/**
 * @member {Number} skip
 */

CompetitionFilter.prototype['skip'] = undefined;
/**
 * @member {Number} limit
 */

CompetitionFilter.prototype['limit'] = undefined;
var _default = CompetitionFilter;
exports["default"] = _default;
