// @ziqni-tech/member-api-client/dist/model/MemberOptinRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The MemberOptinRequest model module.
 * @module model/MemberOptinRequest
 * @version 0.0.1
 */
var MemberOptinRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MemberOptinRequest</code>.
   * @alias module:model/MemberOptinRequest
   * @param memberRefId {String} 
   */
  function MemberOptinRequest(memberRefId) {
    _classCallCheck(this, MemberOptinRequest);

    MemberOptinRequest.initialize(this, memberRefId);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(MemberOptinRequest, null, [{
    key: "initialize",
    value: function initialize(obj, memberRefId) {
      obj['memberRefId'] = memberRefId;
    }
    /**
     * Constructs a <code>MemberOptinRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MemberOptinRequest} obj Optional instance to populate.
     * @return {module:model/MemberOptinRequest} The populated <code>MemberOptinRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MemberOptinRequest();

        if (data.hasOwnProperty('memberRefId')) {
          obj['memberRefId'] = _ApiClient["default"].convertToType(data['memberRefId'], 'String');
        }

        if (data.hasOwnProperty('entityId')) {
          obj['entityId'] = _ApiClient["default"].convertToType(data['entityId'], 'String');
        }

        if (data.hasOwnProperty('entityType')) {
          obj['entityType'] = _ApiClient["default"].convertToType(data['entityType'], 'String');
        }

        if (data.hasOwnProperty('action')) {
          obj['action'] = _ApiClient["default"].convertToType(data['action'], 'String');
        }
      }

      return obj;
    }
  }]);

  return MemberOptinRequest;
}();
/**
 * @member {String} memberRefId
 */


MemberOptinRequest.prototype['memberRefId'] = undefined;
/**
 * @member {String} entityId
 */

MemberOptinRequest.prototype['entityId'] = undefined;
/**
 * @member {String} entityType
 */

MemberOptinRequest.prototype['entityType'] = undefined;
/**
 * @member {String} action
 */

MemberOptinRequest.prototype['action'] = undefined;
var _default = MemberOptinRequest;
exports["default"] = _default;
