// @ziqni-tech/member-api-client/dist/model/ResponseMeta.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The ResponseMeta model module.
 * @module model/ResponseMeta
 * @version 0.0.1
 */
var ResponseMeta = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ResponseMeta</code>.
   * Response metadata   
   * @alias module:model/ResponseMeta
   * @param totalRecordsFound {Number} 
   * @param skip {Number} 
   * @param limit {Number} 
   */
  function ResponseMeta(totalRecordsFound, skip, limit) {
    _classCallCheck(this, ResponseMeta);

    ResponseMeta.initialize(this, totalRecordsFound, skip, limit);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ResponseMeta, null, [{
    key: "initialize",
    value: function initialize(obj, totalRecordsFound, skip, limit) {
      obj['totalRecordsFound'] = totalRecordsFound;
      obj['skip'] = skip;
      obj['limit'] = limit;
    }
    /**
     * Constructs a <code>ResponseMeta</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ResponseMeta} obj Optional instance to populate.
     * @return {module:model/ResponseMeta} The populated <code>ResponseMeta</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ResponseMeta();

        if (data.hasOwnProperty('totalRecordsFound')) {
          obj['totalRecordsFound'] = _ApiClient["default"].convertToType(data['totalRecordsFound'], 'Number');
        }

        if (data.hasOwnProperty('skip')) {
          obj['skip'] = _ApiClient["default"].convertToType(data['skip'], 'Number');
        }

        if (data.hasOwnProperty('limit')) {
          obj['limit'] = _ApiClient["default"].convertToType(data['limit'], 'Number');
        }

        if (data.hasOwnProperty('objectType')) {
          obj['objectType'] = _ApiClient["default"].convertToType(data['objectType'], 'String');
        }
      }

      return obj;
    }
  }]);

  return ResponseMeta;
}();
/**
 * @member {Number} totalRecordsFound
 */


ResponseMeta.prototype['totalRecordsFound'] = undefined;
/**
 * @member {Number} skip
 */

ResponseMeta.prototype['skip'] = undefined;
/**
 * @member {Number} limit
 */

ResponseMeta.prototype['limit'] = undefined;
/**
 * @member {String} objectType
 */

ResponseMeta.prototype['objectType'] = undefined;
var _default = ResponseMeta;
exports["default"] = _default;
