// @ziqni-tech/member-api-client/dist/model/RewardReduced.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
 * The RewardReduced model module.
 * @module model/RewardReduced
 * @version 0.0.1
 */
var RewardReduced = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>RewardReduced</code>.
   * @alias module:model/RewardReduced
   */
  function RewardReduced() {
    _classCallCheck(this, RewardReduced);

    RewardReduced.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(RewardReduced, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>RewardReduced</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/RewardReduced} obj Optional instance to populate.
     * @return {module:model/RewardReduced} The populated <code>RewardReduced</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new RewardReduced();

        if (data.hasOwnProperty('rewardRank')) {
          obj['rewardRank'] = _ApiClient["default"].convertToType(data['rewardRank'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('rewardValue')) {
          obj['rewardValue'] = _ApiClient["default"].convertToType(data['rewardValue'], 'Number');
        }

        if (data.hasOwnProperty('rewardType')) {
          obj['rewardType'] = _ApiClient["default"].convertToType(data['rewardType'], 'String');
        }

        if (data.hasOwnProperty('rewardTypeId')) {
          obj['rewardTypeId'] = _ApiClient["default"].convertToType(data['rewardTypeId'], 'String');
        }
      }

      return obj;
    }
  }]);

  return RewardReduced;
}();
/**
 * If used in the context of contest this will associate with the rank of the leaderboard
 * @member {String} rewardRank
 */


RewardReduced.prototype['rewardRank'] = undefined;
/**
 * The name of a reward
 * @member {String} name
 */

RewardReduced.prototype['name'] = undefined;
/**
 * Numerical value of the reward that will be issued based on the reward type
 * @member {Number} rewardValue
 */

RewardReduced.prototype['rewardValue'] = undefined;
/**
 * Reward Type key
 * @member {String} rewardType
 */

RewardReduced.prototype['rewardType'] = undefined;
/**
 * A unique id of the Reward Type
 * @member {String} rewardTypeId
 */

RewardReduced.prototype['rewardTypeId'] = undefined;
var _default = RewardReduced;
exports["default"] = _default;
